docs/specifications/tests/Methods.md

```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
-5. If the recursive lookup reaches authorititative NOERROR answer, with or without
-   records in the answer section, the method fails, and no parent zone is found.
-6. If the recurse lookup is exhausted or ends in a loop, then the method fails
+   (AA flag set) with NXDOMAIN for the child domain (child zone): 
+   
+   4.1 The method succeeds.
+   
+   4.2 The zone returning NXDOMAIN is considered to be the parent zone.
+   
+   4.3 The non-existance of the child zone has been determined.
+
+   4.4 Return the name of the parent zone and the fact that the child zone does not exist.
+
+5. If the recursive lookup reaches authorititative NOERROR answer (AA flag set), 
+   with or without records in the answer section:
+   
+   5.1 The method succeeds.
+   
+   5.2 The zone returning authoritative data is considered to be the parent zone.
+   
+   4.3 The non-existance of the child zone has been determined.
+
+   4.4 Return the name of the parent zone and the fact that the child zone does not exist.
+   
+6. If the recurse lookup ends with server or zone error or in a loop, then the method fails
    and no parent zone is found.
 
+Parent zone     |Child zone        |Run normal test?|Run undelegated test
+----------------|------------------|----------------|---------------------------------
+Determined      |Exists            |Yes             |Yes (1)
+Determined      |Does not exist (2)|No              |Yes
+Indetermined (3)|Indetermined      |No              |Yes
+
+  (1) Ignore delegation data and used provided data.
+
+  (2) Parent zone returns an authoritative NXDOMAIN, data or nodata on the child zone name.
+  
+  (3) Server or zone error prevents the existence of parent zone to be determined.
+
 
 ## Method 2: Delegation name servers
 
 Obtain the name servers names (extracted from the NS records) for 
 the given zone (child zone) as defined in the delegation from the parent zone.
 
-1. Obtain parent zone using Method 1.
+1. Obtain parent zone and existence of child zone using Method 1.
 
 A. Normal test. For undelegated test go to B.
 
-2. Send an SOA query for the given domain name (zone) to one of the authoritative
-   name servers for the parent zone.
-3. Collect all A and AAAA records in the additional section and store them 
-   in a cache that method 4 can use.
-4. Collect all NS records in the authority section and return the name server names
-   in the response.
+2. If parent zone could not be determined, method fails.
+
+3. If no child zone exists, method fails.
+
+4. Fetch the name server names for child zone from the cache provided by method 1.
+   Return the name server names.
 
 B. Undelegated test.
 
-2. If the test is an undelegated test, then the name servers names in the input
-   will be used as the equivalent of RDATA of the NS records from the lookup 
-   under A.
+2. Ignore if parent zone or child zone does not exist or cannot be determined.
+
+3. Ignore any name servers provided by delegation from parent zone.
```

Comment:
Steps B.2 and B.3 are noops, right? Have you considered moving them to the description?